File: fixtures/typescript/simple.ts
// Fixture: basic class + function + interface + imports.
// The golden file expected/simple.json is the source of truth for extractor output.
import { User } from '../users/User';
import { TokenService } from './TokenService';

export interface AuthConfig {
  timeout: number;
}

export class AuthService {
  constructor(private tokens: TokenService) {}

  login(user: User): string {
    return this.tokens.issue(user);
  }

  logout(token: string): void {
    this.tokens.revoke(token);
  }
}

export function createAuthService(t: TokenService): AuthService {
  return new AuthService(t);
}

export const DEFAULT_TIMEOUT = 5000;
export const handler = (req: unknown) => null;

type Token = string;

File: fixtures/python/simple.py
"""Fixture: basic functions + class + imports, for the Python parser golden test."""
from typing import List
from .token_service import TokenService


class User:
    def __init__(self, name: str):
        self.name = name


class AuthService:
    """Auth service depending on TokenService."""

    def __init__(self, tokens: TokenService):
        self.tokens = tokens

    def login(self, user: User) -> str:
        return self.tokens.issue(user)

    def logout(self, token: str) -> None:
        self.tokens.revoke(token)


def create_auth_service(t: TokenService) -> AuthService:
    return AuthService(t)


DEFAULT_TIMEOUT = 5000
